### ISSUE TITLE
Stack traces are broken on windows

### ISSUE THREAD

**Issue by Sahnvour** [ziglang/zig#1655]
I recently updated my local fork of Zig, and it appears that it can't find source lines anymore when printing a stack trace.

In `printSourceAtAddressWindows`, no section contribution entry is found that matches the addresses, although there are multiple entries and they look correct from what I can tell. WinDBG is able to show the source lines so the .pdb must be correct.

**Comment by andrewrk** [comment]
Looks like it's about time to add a new kind of test to make sure debug info/stack traces are working.

**Comment by emekoi** [comment]
i'm going to do some digging and see if i can find the the commit where is broke.

**Comment by emekoi** [comment]
it think it has something to do with llvm, because i went all through previous commits and stack traces didn't work on any of them.

**Comment by Sahnvour** [comment]
+1, forgot to mention that I did checkout right after the llvm-7.0 merge and it was still broken, but I did not bother to rebuild before it with 6.0.

**Comment by Sahnvour** [comment]
I think I've identified were the problem lies.

By printing all the `SectionContribEntry` we load in the pdb, I noticed in my case that they refer to sections of indices from 1 to 5. But my executable has only 5 COFF sections. Turns out that `SectionIndex` is 1-based cf. https://github.com/llvm-mirror/lld/blob/master/COFF/Writer.h#L60

This index is later used to access a slice element at https://github.com/ziglang/zig/blob/master/std/debug/index.zig#L286. Substracting 1 at this point seems to fix the stack traces.

IIRC, traces worked fine when you finished implementing them for 0.3.0, and I did not see a changeset either in Zig or lld that would have caused this breakage. So I don't understand what really happened.
